c2corg_api/views/document.py

```diff
@@ -7,18 +13,115 @@ class DocumentRest(object):
     def __init__(self, request):
         self.request = request
 
+    def _paginate_offset(self, clazz, schema):
+        offset = self.request.GET.get('offset')
+        limit = self.request.GET.get('limit')
+        offset = 0 if offset is None else int(offset)
+        limit = 30 if limit is None else int(limit)
+
+        base_query = DBSession. \
+            query(clazz)
+
+        documents = base_query. \
+            options(joinedload(getattr(clazz, 'locales'))). \
+            order_by(clazz.document_id). \
+            slice(offset, offset + limit). \
+            all()
+
+        total = base_query.count()
+
+        return {
+            'documents': [to_json_dict(doc, schema) for doc in documents],
+            'total': total
+        }
+
+    def _paginate_after(self, clazz, schema):
+        after = self.request.GET.get('after')
+        limit = self.request.GET.get('limit')
+        after_id = int(after)
+        limit = 30 if limit is None else int(limit)
+
+        base_query = DBSession. \
+            query(clazz)
+
+        documents = base_query. \
+            options(joinedload(getattr(clazz, 'locales'))). \
+            order_by(clazz.document_id). \
+            filter(clazz.document_id > after_id). \
+            limit(limit). \
+            all()
+
+        return {
+            'documents': [to_json_dict(doc, schema) for doc in documents],
+            'total': -1
+        }
+
+    def _paginate(self, clazz, schema):
+        if self.request.GET.get('after') is not None:
+            return self._paginate_after(clazz, schema)
+        else:
+            return self._paginate_offset(clazz, schema)
+
+    def _collection_get(self, clazz, schema):
+        res = self._paginate(clazz, schema)
+        return res['documents']
```

Comment:
Right, computing total count only if not provided in request and return it to client.
The returned json is now:
```
{
  'documents': the_list,
  'total': total count
}
```